spec/micro/README.md

```diff
@@ -0,0 +1,770 @@
+# Micro IBC Architecture
+
+### Context
+
+The implementation of the entire IBC protocol as it currently stands is a large undertaking. While there exists ready-made implementations like ibc-go this is only deployable on the Cosmos-SDK. Similarly, there exists ibc-rs which is a library for chains to integrate. However, this requires the chain to be implemented in Rust, there still exists some non-trivial work to integrate the ibc-rs library into the target state machine, and certain limitations either in the state machine or in ibc-rs may prevent using the library for the target chain.
+
+Writing an implementation from scratch is a problem many ecosystems face as a major barrier for IBC adoption.
+
+The goal of this document is to serve as a "micro-IBC" specification that will allow new ecosystems to implement a protocol that can communicate with fully implemented IBC chains using the same security assumptions.
```

Comment:
I guess that another goal of this document is to revisit the initial design choices, learn from them, keep the core meat, and provide a smoother version of protocol that will be useful not only for new ecosystem, but even for cosmos itself.